WalletWasabi/WabiSabi/Backend/Statistics/CoinJoinFeeRateStatStore.cs

```diff
@@ -67,18 +63,28 @@ private void Add(CoinJoinFeeRateStat feeRateStat)
 		}
 	}
 
-	private FeeRate GetAverage(TimeSpan timeFrame)
+	private FeeRate GetMedian(TimeSpan timeFrame)
 	{
 		var from = DateTimeOffset.UtcNow - timeFrame;
-		return new FeeRate(CoinJoinFeeRateStats.Where(x => x.DateTimeOffset >= from).Average(x => x.FeeRate.SatoshiPerByte));
+		var feeRates = CoinJoinFeeRateStats
+			.Where(x => x.DateTimeOffset >= from)
+			.OrderByDescending(x => x.FeeRate.SatoshiPerByte)
+			.ToArray();
+
+		// If the median is even, then it's the average of the middle two numbers.
+		FeeRate med = feeRates.Length % 2 == 0
+			? new FeeRate((feeRates[feeRates.Length / 2].FeeRate.SatoshiPerByte + feeRates[(feeRates.Length / 2) + 1].FeeRate.SatoshiPerByte) / 2)
```

Comment:
```suggestion
			? new FeeRate((feeRates[feeRates.Length / 2].FeeRate.SatoshiPerByte + feeRates[(feeRates.Length / 2) - 1].FeeRate.SatoshiPerByte) / 2)
```

This will be the middle two elements. Simple example, array of 2 numbers. 2/2 =1 so the indexes should be 0,1 not 1,2.